src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/BotSegments/Models/CVImageMouseActionData.cs

```diff
@@ -0,0 +1,376 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using JetBrains.Annotations;
+using Newtonsoft.Json;
+using RegressionGames.StateRecorder.BotSegments.JsonConverters;
+using RegressionGames.StateRecorder.BotSegments.Models.CVService;
+using RegressionGames.StateRecorder.JsonConverters;
+using RegressionGames.StateRecorder.Models;
+using StateRecorder.BotSegments.Models;
+using UnityEngine;
+using Random = UnityEngine.Random;
+
+namespace RegressionGames.StateRecorder.BotSegments.Models
+{
+    public class CVMouseActionDetails
+    {
+        public static readonly int apiVersion = SdkApiVersion.VERSION_10;
+
+        //main 5 buttons
+        public bool leftButton;
+        public bool middleButton;
+        public bool rightButton;
+        public bool forwardButton;
+        public bool backButton;
```

Comment:
I think it could be a useful feature to have a mouse position offset from the recognized CV image be an optional parameter of the actions, e.g. to aid in specifying click->drag actions.

Reply:
I started with something like that, but for now decided that a click drag (at least for PW) would be handled as 2 segment files.  One with the mouse down on the first detected image, and a second with a mouse up on the destination detected image.